aws/resource_aws_opsworks_stack_test.go

```diff
@@ -236,24 +236,41 @@ func testAccCheckAWSOpsworksStackRecreated(t *testing.T, before, after *opsworks
 	}
 }
 
+func testAccPreCheckAWSOpsWorksStacks(t *testing.T) {
+	conn := testAccProvider.Meta().(*AWSClient).opsworksconn
+
+	input := &opsworks.DescribeStacksInput{}
+
+	_, err := conn.DescribeStacks(input)
+
+	if testAccPreCheckSkipError(err) {
+		t.Skipf("skipping acceptance testing: %s", err)
+	}
+
+	if err != nil {
+		t.Fatalf("unexpected PreCheck error: %s", err)
+	}
+}
+
 func testAccAwsOpsWorksStack_classic_endpoint(rName string, rInt int) string {
 	return fmt.Sprintf(`
 provider "aws" {
-  region = "us-east-1"
```

Comment:
The history behind this pre-dates me by awhile, but this resource has special handling for `us-east-1`:

https://github.com/terraform-providers/terraform-provider-aws/blob/fd2167593e3ac0678512e32da9ac90b2edf83741/aws/resource_aws_opsworks_stack.go#L293-L318

Unsure if this hardcoded region can be updated in this particular test (we'll need to go back in the git history to see which test was written for this logic). Whatever that yields, we may just want to gate any special `us-east-1` testing to the `aws` partition. 👍

Reply:
Wow. Going through history can be scary. There are two related but separate issues:

1. At one point, OpsWorks only had a `us-east-1` endpoint. By TF 0.9, OpsWorks supported regional endpoints and TF support was added. The special handling in the resource exists for backward compatibility - earlier configs had no region. No test was added for this.
2. With the introduction of regional endpoints, you could create an OpsWorks Stack from a region but have it live in a different region. It could only (can only?) be managed from the creation region, not the region it lives in. The `TestAccAWSOpsWorksStack_classic_endpoints` (since renamed) test was added to address this issue. This test isn't about anything specific to `us-east-1`.

Reply:
Thank you for walking back through that one. Sounds like we're good to update this like you did. 👍